agent/consul/acl_endpoint.go

```diff
@@ -26,9 +26,34 @@ const (
 // Regex for matching
 var validPolicyName = regexp.MustCompile(`^[A-Za-z0-9\-_]{1,128}$`)
 
+const (
+	// maxTokenExpirationDuration is the maximum difference allowed between
+	// ACLToken CreateTime and ExpirationTime values.
+	maxTokenExpirationDuration = 24 * time.Hour
+
+	// minTokenExpirationDuration is the minimum difference allowed between
+	// ACLToken CreateTime and ExpirationTime values.
+	minTokenExpirationDuration = 1 * time.Minute
+)
+
 // ACL endpoint is used to manipulate ACLs
 type ACL struct {
 	srv *Server
+
+	// clock is a way to control access to the current time. This is mainly
+	// here for use in tests. Not all time accesses go through this, so use it
+	// from tests with care.
+	//
+	// This clock should only really be used for correctness-related operations
+	// and not for diagnostic metrics timing.
+	clock clock
```

Comment:
I did start to do it that way twice but either got hung up on gross plumbing details, or in a vague nagging feeling that I "should" thread it through everything that used `time.Now` in the whole app.  I guess the comment block I already provide is sufficiently disclaimery that it could ride just fine back up in the `Server`.